### PR TITLE
[INLONG-7273][Manager] Support creating table in Kudu cluster

### LINKED ISSUE #7273: [Feature][Manager] Support creating table in Kudu cluster
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by featzhang**
### Prepare a Pull Request

- [INLONG-7273][Manager] Support creating table in Kudu cluster

- Fixes #7273 

image

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/kudu/KuduResourceClient.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.util.List;
+
+public class KuduResourceClient {
+
+    public KuduResourceClient(String metastoreUri, String dbName) {
+
+    }
+
+    public void open() {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/kudu/KuduResourceClient.java:31]: Excuse me, but why those methods were empty?

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/kudu/KuduResourceOperator.java [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    private DataNodeOperateHelper dataNodeHelper;
+
+    @Override
+    public Boolean accept(String sinkType) {
+        return SinkType.KUDU.equals(sinkType);
+    }
+
+    /**
+     * Create Kudu table according to the sink config
+     */
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/kudu/KuduResourceOperator.java:66]: @Override

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/kudu/KuduResourceClient.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            String desc = columnInfo.getFieldComment();
+
+            String kuduType = KuduType.forType(type).kuduType();
+            Type typeForName = Type.getTypeForName(kuduType);
+
+            ColumnSchema columnSchema = buildColumnSchema(name, desc, typeForName);
+
+            alterOptions.addColumn(columnSchema);
+        }
+        alterOptions.addColumn("new_column_name", Type.STRING, "default_value");
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/kudu/KuduResourceClient.java:165]: Excuse me, what is the function of this line?

**Review by healchow — APPROVED** [review]
LGTM

**Review by dockerzhang — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/kudu/KuduResourceClient.java [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+    public KuduResourceClient(String kuduMaster) {
+        this.client = new KuduClient.KuduClientBuilder(kuduMaster).build();
+    }
+
+    public boolean tableExist(String tableName) {
+        try {
+            return client.tableExists(tableName);
+        } catch (KuduException e) {
+            LOG.error("Can not properly query kudu!", e);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/sink/kudu/KuduResourceClient.java:60]: Why not throw an exception here?

**Review by vernedeng — APPROVED** [review]
LGTM

**Review by healchow — APPROVED** [review]

**Pull request merged**